File: scripts/generate_manifest.js
const fs = require('fs');
const https = require('https');

const GANJOOR_API = "https://api.ganjoor.net/api/ganjoor";

const POET_IDS = {
    "Rudaki": 1,
    "Saadi": 7,
    "Nizami": 22,
    "Rumi": 5,
    "Hafiz": 2
};

function fetchPoetMetadata(poetId) {
    return new Promise((resolve, reject) => {
        https.get(`${GANJOOR_API}/poet/${poetId}`, (res) => {
            let data = '';
            res.on('data', (chunk) => { data += chunk; });
            res.on('end', () => {
                try {
                    resolve(JSON.parse(data));
                } catch (e) {
                    reject(e);
                }
            });
        }).on('error', (e) => reject(e));
    });
}

async function main() {
    const metadata = {};
    for (const [name, pid] of Object.entries(POET_IDS)) {
        console.log(`Fetching ${name}...`);
        try {
            const data = await fetchPoetMetadata(pid);
            if (data && data.poet) {
                metadata[name] = {
                    "ganjoor_id": data.poet.id,
                    "name": data.poet.name,
                    "nickname": data.poet.nickname,
                    "root_cat_id": data.poet.rootCatId,
                    "birth_place": data.poet.birthPlace || "Unknown",
                    "death_place": data.poet.deathPlace || "Unknown"
                };
            }
        } catch (e) {
            console.error(`Error fetching poet ${pid}: ${e.message}`);
        }
    }
    
    if (!fs.existsSync('data')) fs.mkdirSync('data');
    fs.writeFileSync('data/poets_manifest.json', JSON.stringify(metadata, null, 2), 'utf8');
    console.log("Manifest created in data/poets_manifest.json");
}

main();

File: scripts/generate_manifest.py
import requests
import json
import sys

GANJOOR_API = "https://api.ganjoor.net/api/ganjoor"

# Initial priority poets for the Silk Road Synthesis
POET_IDS = {
    "Rudaki": 1,
    "Saadi": 7,
    "Nizami": 22,
    "Rumi": 5,
    "Hafiz": 2
}

def fetch_poet_metadata(poet_id):
    try:
        response = requests.get(f"{GANJOOR_API}/poet/{poet_id}")
        response.raise_for_status()
        return response.json()
    except Exception as e:
        print(f"Error fetching poet {poet_id}: {e}")
        return None

def main():
    metadata = {}
    for name, pid in POET_IDS.items():
        print(f"Fetching {name}...")
        data = fetch_poet_metadata(pid)
        if data:
            # We only need the core poet info for the manifest
            metadata[name] = {
                "ganjoor_id": data["poet"]["id"],
                "name": data["poet"]["name"],
                "nickname": data["poet"]["nickname"],
                "root_cat_id": data["poet"]["rootCatId"],
                "birth_place": data["poet"].get("birthPlace", "Unknown"),
                "death_place": data["poet"].get("deathPlace", "Unknown")
            }
    
    with open("data/poets_manifest.json", "w", encoding="utf-8") as f:
        json.dump(metadata, f, ensure_ascii=False, indent=2)
    print("Manifest created in data/poets_manifest.json")

if __name__ == "__main__":
    main()
